src/main/java/com/exactpro/th2/readcsv/Main.java

```diff
@@ -189,8 +260,21 @@ private static RawMessage.Builder validateAndAppend(
         return it.setBody(extractedHeader.getContent().concat(it.getBody()));
     }
 
+    private static com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage.Builder transportValidateAndAppend(
+            HeaderHolder<ByteBuf> headerHolder,
+            HeaderInfo<ByteBuf> extractedHeader,
+            com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage.Builder it,
+            boolean validate,
+            boolean validateOnlyExtraData
+    ) {
+        if (validate) {
+            headerHolder.validateContentSize(extractedHeader, it.getBody(), validateOnlyExtraData);
+        }
+        return it.setBody(extractedHeader.getContent().writeBytes(it.getBody()));
```

Comment:
If I see it correctly, we modify the `extractedHeader.getContent()`. That means only the first execution will be correct. After that, the `extractedHeader.getContent()` will have the content of a previous message as well. We must handle this correctly and avoid headers ByteBuf mutation...